### PR TITLE
fix: 修复 2k 显示器切换到 MacBook 后表格渲染模糊 close #2072

### LINKED ISSUE #2072: 🐛 外接显示器拖到 MacBook 表格渲染模糊
### 🏷 Version

| Package        | Version |
| -------------- | ------- |
| @antv/s2       |    1.41.0     |
| @antv/s2-react |         |
| @antv/s2-vue   |         |

### Sheet Type

- [x] PivotSheet
- [x] TableSheet
- [ ] GridAnalysisSheet
- [ ] StrategySheet
- [ ] EditableSheet

### 🖋 Description

image

原因: DPR 改变时 `visualViewport` 触发了 `resize` 事件

image

导致 DPR 从 1 => 2 => 1  额外渲染了一次, DPR 被重置了

### 💻 System information

| Environment | Info |
| ------- | ------- |
| System  |         |
| Browser |         |

### PR DESCRIPTION
**Pull request by lijinke666**
### 👀 PR includes

✨ Feature

- [ ] New feature

🎨 Enhance

- [ ] Code style optimization
- [ ] Refactoring
- [ ] Change the UI
- [ ] Improve the performance
- [ ] Type optimization

🐛 Bugfix

- [x] Solve the issue and close #2072 

### 📝 Description

背景见 #2072 

原因: 我们通过 `matchMedia` 监听设备 DPR 的改变来做高清渲染, 通过 `visualViewport` 的 `resize` 来监听Mac 触摸板双指缩放, DPR改变时 `resize` 事件也会触发 (在 matchMedia) 之后, 导致 DPR 从 1 => 2 => 1 , 多渲染了一次

### 🖼️ Screenshot

> 放大查看更明显

| Before | After |
| ------ | ----- |
| Kapture 2023-02-13 at 20 07 20 | Kapture 2023-02-13 at 20 05 24   |
| image | image |

### 🔍 Self-Check before the merge

- [ ] Add or update relevant docs.
- [ ] Add or update relevant demos.
- [ ] Add or update test case.
- [ ] Add or update relevant TypeScript definitions.

### CODE REVIEW

File: packages/s2-core/src/ui/hd-adapter/index.ts [hunk 1]
```diff
@@ -107,7 +121,7 @@ export class HdAdapter {
   private renderByZoomScale = debounce(
     (event: Event & { target: VisualViewport }) => {
       const ratio = Math.ceil(event.target.scale);
-      if (ratio >= 1) {
+      if (ratio >= 1 && !this.isDevicePixelRatioChange) {
```
Comment (wjgogogo) [inline packages/s2-core/src/ui/hd-adapter/index.ts:124]: isDevicePixelRatioChange 在执行完了后需要重置不？

Comment (lijinke666) [reply packages/s2-core/src/ui/hd-adapter/index.ts:124]: 在缩放的时候会重置 
image

**Review by wjgogogo — APPROVED** [review]

**Pull request merged**